### PR TITLE
REF: Delegate more methods for DTI/TDI/PI

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -66,8 +67,13 @@ class DatetimeDelegateMixin(DatetimelikeDelegateMixin):
     # We also have a few "extra" attrs, which may or may not be raw,
     # which we we dont' want to expose in the .dt accessor.
     _extra_methods = ["to_period", "to_perioddelta", "to_julian_date", "strftime"]
-    _extra_raw_methods = ["to_pydatetime", "_local_timestamps", "_has_same_tz"]
-    _extra_raw_properties = ["_box_func", "tz", "tzinfo"]
+    _extra_raw_methods = [
```
Comment (jreback) [inline pandas/core/indexes/datetimes.py]: see my comment on the other PR, I don't find _raw_ very intuitive here (maybe _delegated?)

Comment (jreback) [reply pandas/core/indexes/datetimes.py]: ahh we *already* have _delegated_methods and so on, ok, i think we still need a better name, how would you describe these properties?

File: pandas/core/arrays/timedeltas.py [hunk 2]
```diff
@@ -445,7 +445,7 @@ def _formatter(self, boxed=False):
 
         return _get_format_timedelta64(self, box=True)
 
-    def _format_native_types(self, na_rep="NaT", date_format=None):
+    def _format_native_types(self, na_rep="NaT", date_format=None, **kwargs):
```
Comment (jreback) [inline pandas/core/arrays/timedeltas.py]: do we *actually* use these kwargs (i know they are on the method you are removing)?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/datetimes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/datetimes.py]: yah, like i mentioned there im planning a significant cleanup of these

File: pandas/core/arrays/timedeltas.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/timedeltas.py]: yes, without this we get a test failure bc `float_format` is passed

Comment (jreback) [reply pandas/core/arrays/timedeltas.py]: ahh i c, as these are fairly generic, ok then.

**Review by jreback — APPROVED** [review]

**Pull request merged**